Neo4j.Driver/Neo4j.Driver.IntegrationTests/RoutingDriver/RoutingDriverStressIT.cs

```diff
@@ -0,0 +1,213 @@
+
+            Task.WaitAll(tasks.ToArray());
+            _cancellationTokenSource.Cancel();
+
+            var endTime = DateTime.Now;
+            Output.WriteLine($"[{endTime:HH:mm:ss.ffffff}] Finished");
+            Output.WriteLine($"Total time spent: {endTime - startTime}");
+
+            PrintStatistics();
+
+            _driver.Close();
+        }
+
+        [RequireClusterTheory]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
+        public async Task RoutingDriverStressTestAsync(int queryCount)
+        {
+            const int taskCount = 5;
+            var startTime = DateTime.Now;
+            Output.WriteLine($"[{startTime:HH:mm:ss.ffffff}] Started");
+
+            var workItem = new SoakRunWorkItem(_driver, _metrics, Output);
+
+            var terminationTask = ConnectionTerminatorAsync();
+            var tasks = new List<Task>();
+            for (var i = 0; i < taskCount; i++)
+            {
+                tasks.Add(workItem.RunWithRetriesAsync(queryCount / taskCount));
+            }
+
+            await Task.WhenAll(tasks);
+            _cancellationTokenSource.Cancel();
+            await terminationTask;
+
+            var endTime = DateTime.Now;
+            Output.WriteLine($"[{endTime:HH:mm:ss.ffffff}] Finished");
+            Output.WriteLine($"Total time spent: {endTime - startTime}");
+
+            PrintStatistics();
+
+            await _driver.CloseAsync();
+        }
+
+        private void PrintStatistics()
+        {
+            var poolMetrics = _metrics.PoolMetrics;
+            Output.WriteLine(poolMetrics.ToContentString());
+
+            foreach (var value in poolMetrics)
+            {
+                var st = value.Value;
+
+                st.ToCreate.Should().Be(0);
+                st.ToClose.Should().Be(0);
+                st.InUse.Should().Be(0);
+                st.Idle.Should().Be((int) (st.Created + st.FailedToCreate - st.Closed));
+            }
+        }
+
+        private void SetupMonitoredDriver()
+        {
+            var config = new Config
+            {
+                DriverMetricsEnabled = true,
+                ConnectionAcquisitionTimeout = TimeSpan.FromMinutes(5),
+                ConnectionTimeout = Config.InfiniteInterval,
+                MaxConnectionPoolSize = 50,
+                Logger = new TestLogger(Output)
+            };
+
+            var connectionSettings = new ConnectionSettings(AuthToken, config);
+            var bufferSettings = new BufferSettings(config);
+            var connectionFactory = new MonitoredPoolledConnectionFactory(
+                new PooledConnectionFactory(connectionSettings, bufferSettings, config.Logger));
+
+            _driver = (Internal.Driver) GraphDatabase.CreateDriver(new Uri(RoutingServer), config, connectionFactory);
+            _connections = connectionFactory.Connections;
+            _metrics = _driver.GetDriverMetrics();
+        }
+
+        private Task ConnectionTerminator()
```

Comment:
This killing log is to simulate there is something wrong on the connections between the server and client. The aim is to let the driver realize a connection error and remove the server from routing table and purge idle in connection pool, but the query should still success as we are running queries in retries.

The `MaxIdle` is not set on purpose so that it will default to the same size as `MaxConn`.